store/postgres/src/chain_store.rs

```diff
@@ -1877,6 +1954,42 @@ mod recent_blocks_cache {
             }
         }
 
+        fn record_cache_hit(&self) {
+            self.metrics
+                .chain_head_cache_hits
+                .get_metric_with_label_values(&[&self.network])
+                .unwrap()
+                .inc();
+        }
+
+        fn record_cache_miss(&self) {
+            self.metrics
+                .chain_head_cache_misses
+                .get_metric_with_label_values(&[&self.network])
+                .unwrap()
+                .inc();
+        }
+
+        fn update_write_metrics(&self) {
+            self.metrics
+                .chain_head_cache_size
+                .get_metric_with_label_values(&[&self.network])
+                .unwrap()
+                .set(self.blocks.len() as f64);
+
+            self.metrics
+                .chain_head_cache_oldest_block_num
+                .get_metric_with_label_values(&[&self.network])
+                .unwrap()
+                .set(self.earliest_block().map(|b| b.ptr.number).unwrap_or(0) as f64);
+
+            self.metrics
+                .chain_head_cache_latest_block_num
+                .get_metric_with_label_values(&[&self.network])
+                .unwrap()
+                .set(self.earliest_block().map(|b| b.ptr.number).unwrap_or(0) as f64);
```

Comment:
This is using `earliest_block()`, shouldn't it be using `chain_head()`?

Reply:
Good catch!